src/include/kdbprivate.h

```diff
@@ -539,6 +539,54 @@ int elektraKeyLock (Key * key, enum elektraLockOptions what);
 
 ssize_t ksSearchInternal (const KeySet * ks, const Key * toAppend);
 
+/**
```

Comment:
whats missing here is the discussed `#ifdef` to avoid Elektra getting larger for embedded systems. Maybe even go for an extra header file?